pkg/server/commands/reverseexpand/reverse_expand_test.go

```diff
@@ -443,3 +447,277 @@ func TestReverseExpandIgnoresInvalidTuples(t *testing.T) {
+				Throttler:    mockThrottler,
+				Threshold:    200,
+				MaxThreshold: 200,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(0)
+		dispatchCountValue := uint32(190)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("above_threshold_should_call_throttle", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    200,
+				MaxThreshold: 200,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(1)
+		dispatchCountValue := uint32(201)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("zero_max_should_interpret_as_default", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    200,
+				MaxThreshold: 0,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(0)
+		dispatchCountValue := uint32(190)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("dispatch_should_use_request_threshold_if_available", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    0,
+				MaxThreshold: 210,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(1)
+		dispatchCountValue := uint32(201)
+		ctx := context.Background()
+		ctx = threshold.ContextWithDispatchThrottlingThreshold(ctx, 200)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+
+	t.Run("should_respect_max_threshold", func(t *testing.T) {
+		mockThrottler := mocks.NewMockThrottler(mockController)
+		reverseExpandQuery := NewReverseExpandQuery(
+			mockDatastore,
+			ts,
+			WithDispatchThrottlerConfig(threshold.Config{
+				Throttler:    mockThrottler,
+				Threshold:    200,
+				MaxThreshold: 300,
+			}),
+		)
+		mockThrottler.EXPECT().Throttle(gomock.Any()).Times(1)
+		dispatchCountValue := uint32(301)
+		ctx := context.Background()
+		ctx = threshold.ContextWithDispatchThrottlingThreshold(ctx, 1000)
+		metadata := NewResolutionMetadata()
+		metadata.DispatchCounter.Store(dispatchCountValue)
+
+		reverseExpandQuery.throttle(ctx, metadata)
+	})
+}
+
+func TestReverseExpandDispatchCount(t *testing.T) {
```

Comment:
This test is flaky 

```
~/GitHub/openfga git:[double-queue-list-objects-dispatch-throttling]
go test ./pkg/server/commands/reverseexpand/ -race -count=100 -run="TestReverseExpand"
--- FAIL: TestReverseExpandDispatchCount (0.00s)
    reverse_expand.go:627: Unexpected call to *mocks.MockThrottler.Throttle([context.Background.WithValue(type storage.ctxKey, val <not Stringer>).WithValue(type typesystem.ctxKey, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type graph.ctxKey, val <not Stringer>).WithCancel.WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithCancel]) at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand.go:627 because: 
        expected call at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand_test.go:690 has already been called the max number of times
--- FAIL: TestReverseExpandDispatchCount (0.00s)
    reverse_expand.go:627: Unexpected call to *mocks.MockThrottler.Throttle([context.Background.WithValue(type storage.ctxKey, val <not Stringer>).WithValue(type typesystem.ctxKey, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type graph.ctxKey, val <not Stringer>).WithCancel.WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithCancel.WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type graph.ctxKey, val <not Stringer>)]) at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand.go:627 because: 
        expected call at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand_test.go:690 has already been called the max number of times
--- FAIL: TestReverseExpandDispatchCount (0.00s)
    reverse_expand.go:627: Unexpected call to *mocks.MockThrottler.Throttle([context.Background.WithValue(type storage.ctxKey, val <not Stringer>).WithValue(type typesystem.ctxKey, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type graph.ctxKey, val <not Stringer>).WithCancel.WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithCancel]) at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand.go:627 because: 
        expected call at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand_test.go:690 has already been called the max number of times
--- FAIL: TestReverseExpandDispatchCount (0.00s)
    reverse_expand.go:627: Unexpected call to *mocks.MockThrottler.Throttle([context.Background.WithValue(type storage.ctxKey, val <not Stringer>).WithValue(type typesystem.ctxKey, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type graph.ctxKey, val <not Stringer>).WithCancel.WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type trace.traceContextKeyType, val <not Stringer>).WithCancel.WithValue(type trace.traceContextKeyType, val <not Stringer>).WithValue(type graph.ctxKey, val <not Stringer>)]) at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand.go:627 because: 
        expected call at /Users/maria.inesparnisari/GitHub/openfga/pkg/server/commands/reverseexpand/reverse_expand_test.go:690 has already been called the max number of times
FAIL
FAIL    github.com/openfga/openfga/pkg/server/commands/reverseexpand    1.472s
FAIL

```